agent/agent.go

```diff
@@ -134,20 +138,33 @@ func New(config *Config) (Agent, error) {
 		return nil, trace.Wrap(err, "failed to serve prometheus metrics")
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), timelineInitTimeout)
+	defer cancel()
+
+	timeline, err := sqlite.NewTimeline(ctx, sqlite.Config{
+		DBPath:            config.DBPath,
+		RetentionDuration: config.RetentionDuration,
+		Clock:             clockwork.NewRealClock(),
```

Comment:
Nit: The way we usually do it is for things like that we don't pass them explicitly but have a CheckAndSetDefaults method on the Config struct that sets it to the default value (real clock in this case) if it wasn't specified. For example:

```go
type Config {
  Clock clockwork.Clock
}

func (c *Config) CheckAndSetDefaults() error {
  if c.Clock == nil {
    c.Clock = clockwork.NewRealClock()
  }
  return nil
}

func NewTimeline(ctx context.Context, config Config) (xxx, error) {
  if err := config.CheckAndSetDefaults(); err != nil {
    return nil, trace.Wrap(err)
  }
  ...
}

timeline, err := sqlite.NewTimeline(ctx, sqlite.Config{
  DBPath: xxx,
  RetentionDuration: yyy,
})
```

This way the caller doesn't have to pass it explicitly but the tests can still override it.